arrow/arrow.py

```diff
@@ -117,7 +125,10 @@ def __init__(self, stoichiometry, rates, random_seed=0):
         self.obsidian = obsidian.obsidian(
             self.random_seed,
             self.stoichiometry,
-            self.rates,
+            self.rates_flat,
+            self.rates_lengths,
+            self.rates_indexes,
+            self.forms,
```

Comment:
More args! Let's talk about what it looks like to make structs for the nested lists.... may be more trouble than it is worth, but could also clarify some things. It is a tradeoff we should evaluate.